cc_ggplot2_cheatsheet.Rmd

```diff
@@ -0,0 +1,170 @@
+
+
+## Many variables
+```{r}
+mtcars$fcyl_fam <- interaction(mtcars$cyl,mtcars$am,sep = ":")
+
+## Update the plot
+ggplot(mtcars, aes(x = wt, y = mpg, color = fcyl_fam, size = disp)) +
+  geom_point() +
+  scale_color_brewer(palette = "Paired") +
+  ## Grid facet on gear and vs
+  facet_grid(rows = vars(gear), cols = vars(vs))
+```
+
+
+## Formula notation
+```{r}
+ggplot(mtcars, aes(wt, mpg)) + 
+  geom_point() +
+  ## Facet rows by am using formula notation
+  facet_grid(am ~ .)
+
+ggplot(mtcars, aes(wt, mpg)) + 
+  geom_point() +
+  ## Facet columns by cyl using formula notation
+  facet_grid(. ~ cyl)
+
+ggplot(mtcars, aes(wt, mpg)) + 
+  geom_point() +
+  ## Facet rows by am and columns by cyl using formula notation
+  facet_grid(am ~ cyl)
+```
+
+## Labeling facets
+```{r}
+## Plot wt by mpg
+ggplot(mtcars, aes(wt, mpg)) +
+  geom_point() +
+  ## Displaying both the values and the variables
+  facet_grid(cols = vars(cyl), labeller = label_both)
+
+## Plot wt by mpg
+ggplot(mtcars, aes(wt, mpg)) +
+  geom_point() +
+  ## Label context
+  facet_grid(cols = vars(cyl), labeller = label_context)
+```
+
+## Coordinates
+
+
+## Expand and clip
+```{r}
+## Expand sets a buffer margin around the plot, so data and axes don't overlap. 
+## Setting expand to 0 draws the axes to the limits of the data.
+## Clip decides whether plot elements that would lie outside the plot panel 
+## are displayed or ignored ("clipped").
+
+ggplot(mtcars, aes(wt, mpg)) +
+  geom_point(size = 2) +
+  ## Add Cartesian coordinates with zero expansion
+  coord_cartesian(expand = 0) +
+  theme_classic()
+
+ggplot(mtcars, aes(wt, mpg)) +
+  geom_point(size = 2) +
+  ## Turn clipping off
+  coord_cartesian(expand = 0, clip = "off") +
+  theme_classic() +
+  ## Remove axis lines
+  theme(axis.line = element_blank())
+```
+
+
+## Flipping axes 
+```{r}
+## Plot fcyl bars, filled by fam
+ggplot(mtcars, aes(as.factor(cyl), fill = as.factor(am))) +
+  ## Place bars side by side
+  geom_bar(position = "dodge")
+
+ggplot(mtcars, aes(as.factor(cyl), fill = as.factor(am))) +
+  geom_bar(position = "dodge") +
+  ## Flip the x and y coordinates
+  coord_flip()
+
+ggplot(mtcars, aes(as.factor(cyl), fill = as.factor(am))) +
+  ## Set a dodge width of 0.5 for partially overlapping bars
+  geom_bar(position = position_dodge(width = 0.5)) +
+  coord_flip()
+```
+
+
+## Pie charts
+```{r}
+## Run the code, view the plot, then update it
+ggplot(mtcars, aes(x = 1, fill = as.factor(cyl))) +
+  geom_bar()+
+  ## Add a polar coordinate system
+  coord_polar(theta = "y")
+
+ggplot(mtcars, aes(x = 1, fill = as.factor(cyl))) +
+  ## Reduce the bar width to 0.1
+  geom_bar(width = 0.1) +
+  coord_polar(theta = "y") +
+  ## Add a continuous x scale from 0.5 to 1.5
+  scale_x_continuous(limits = c(0.5,1.5))
+```
+
+## Conclusion
+## This is only a little part of ggplot2, and then in my additional resources, compared with ggplot2, 
+## everyone can learn the visualization tools from Python if needed. 
```

Comment:
Are these double hashtags intentional? If not, double hashtags are only needed for Conclusion heading and can be removed from lines 144-145.

Reply:
I did it right now. Thank you!